qcodes/dataset/exporters/export_info.py

```diff
@@ -0,0 +1,34 @@
+"""
+This module defines the ExportInfo dataclass
+"""
+import json
+from dataclasses import asdict, dataclass
+from typing import Dict
+
+from qcodes.dataset.export_config import DataExportType
+
+
+@dataclass
+class ExportInfo:
+
+    export_paths: Dict[str, str]
+
+    def __post_init__(self) -> None:
+        allowed_keys = tuple(a.value for a in DataExportType)
+        for key in self.export_paths.keys():
+            if key not in allowed_keys:
+                raise TypeError(
```

Comment:
instead of raising, shall we warn? or warn and skip the unrecognized keys? in case qcodes version A supports export to X, and version qcodes B supports export X and Y, not-raising-an-exception would allow one to open a dataset in qcodes A using method X that was exported with method X and method Y using qcodes B -- because qcodes A would see the method Y that is not known to it but the code will still work because it will actually use the method X that it supports.

Reply:
Hmn that is worth considering. Originally I wanted the enum to be part of the type key but that is hard to json serialize